firmware/application/apps/ui_recon.cpp

```diff
@@ -1377,10 +1384,31 @@ void ReconView::start_repeat() {
         repeat_file_error(rawfile, "Can't open file to send to thread");
         return;
     }
-    repeat_ready_signal = true;
-    repeat_cur_rep++;
+    // wait for TX if needed (hackish, direct screen update since the UI will be blocked)
+    if (persistent_memory::recon_repeat_delay() > 0) {
+        uint8_t delay = persistent_memory::recon_repeat_delay();
+        Painter p;
+        while (delay > 0) {
+            std::string delay_message = "TX DELAY: " + to_string_dec_uint(delay) + "s";
+
+            // update display information
+            p.fill_rectangle({0, (SCREEN_H / 2) - 16, SCREEN_W, 64}, Color::light_grey());
+            p.draw_string({(SCREEN_W / 2) - 7 * 8, SCREEN_H / 2}, Styles::red, delay_message);
+
+            // sleep 1 second
+            chThdSleepMilliseconds(1000);
```

Comment:
how doesn't this freeze the ui? :)

Reply:
It is freezing the UI hence I'm drawing on screen directly

Reply:
I just didn't wanted to leave the users without an hint on what's going on.
Being able to display a timer was enough since the tx coming after can be interrupted

Reply:
I see

Reply:
:-)